esp-hal/src/parl_io.rs

```diff
@@ -1758,9 +1758,8 @@ pub mod asynch {
                 }
             }
 
-            let future = DmaRxDoneChFuture::new(&mut self.rx_channel);
-            Self::start_receive_bytes_dma(future.rx, &mut self.rx_chain, ptr, len)?;
-            future.await?;
+            Self::start_receive_bytes_dma(&mut self.rx_channel, &mut self.rx_chain, ptr, len)?;
+            DmaRxDoneChFuture::new(&mut self.rx_channel).await?;
```

Comment:
While this works, I wonder what made it not work before? Probably I missed something but I haven't seen anything modifying INT_ENA in `start_receive_bytes_dma` (and functions called by it)

Reply:
Yeah that's what I'm wondering. Maybe something turns of the interrupt, I'll do some more poking.

Reply:
My current suspicion is the interrupt handler. It may make sense to have a separate handler per future actually but I don't want to go that far in this PR.
I'll just apply the enhancement early to move the INT_ENA setting into the poll method.

Reply:
It appears there's a bug in the PARL_IO driver.

`read_dma_async` shouldn't be using `DmaRxDoneChFuture` as that only waits for a single descriptor to be processed. (The example receives 32000 bytes which requires more than one descriptor.)

What's actually happening is `read_dma_async` returns before the transfer is done, the next time it's called the future is created (enabling the interrupts), interrupt fires (waking the waker and disabling the interrupt), new transfer begins (clearing the interrupt bit as well), future is awaited forever since the enable bit was disabled.

Reply:
oh you are right it should use `DmaRxFuture`

Reply:
I created #1841 for this - I think this PR is fine now